apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -16,14 +19,39 @@ const UserChatMessageEditor: React.FunctionComponent = () => {
     (state: {aichat: AichatState}) => state.aichat.isWaitingForChatResponse
   );
 
-  // TODO: If systemPrompt is undefined, handle this error case.
+  const userId: number = useSelector(
+    (state: {currentUser: CurrentUserState}) => state.currentUser.userId
+  );
+
+  const {currentLevelId, scriptId} = useSelector(
+    (state: {progress: ProgressState}) => state.progress
+  );
+
+  const channelId = getStandaloneProjectId();
```

Comment:
This should come from lab2Redux (`state.lab.channel.id`). the `standaloneProjectId` only exists for standalone projects (/projects/aichat) and we only fetch it when first loading the page because we need to grab it out of app options.

Reply:
Will assign with `useSelector` but okay to leave in this component?